app/core/edge_inference.py

```diff
@@ -62,14 +62,48 @@ def parse_inference_response(response: dict) -> dict:
     #     "predictions": {"confidences": [0.54], "labels": [0], "probabilities": [0.45], "scores": [-2.94]},  # Binary results
     #     "secondary_predictions": None,  # Text recognition and Obj detection results
     # }
-    output_dict = response["predictions"]
-    output_dict = {
-        "score": output_dict["scores"][0],
-        "confidence": output_dict["confidences"][0],
-        "probability": output_dict["probabilities"][0],
-        "label": output_dict["labels"][0],
-    }
-    output_dict["label"] = "NO" if output_dict["label"] else "YES"  # map false / 0 to "YES" and true / 1 to "NO"
+
+    multi_predictions: dict = response.get("multi_predictions", None)
+    predictions: dict = response.get("predictions", None)
+    secondary_predictions: dict = response.get("secondary_predictions", None)
+
+    if multi_predictions is not None and predictions is not None:
+        raise ValueError("Got result with both multi_predictions and predictions.")
+    if multi_predictions is not None:
+        # Count or multiclass case
+        probabilities: list[float] = multi_predictions["probabilities"][0]
+        confidence: float = max(probabilities)
+        max_prob_index = max(range(len(probabilities)), key=lambda i: probabilities[i])
+        label: int = max_prob_index
+    elif predictions is not None:
+        # Binary case
+        confidence: float = predictions["confidences"][0]
+        label: int = predictions["labels"][0]
+    else:
+        raise ValueError("Got result with no multi_predictions or predictions.")
+
+    rois: list[dict] | None = None
+    text: str | None = None
+    # Attempt to extract rois / text
+    if secondary_predictions is not None:
+        roi_predictions: list[list[dict]] | None = secondary_predictions.get("roi_predictions", None)
+        text_predictions: list[str] | None = secondary_predictions.get("text_predictions", None)
+        if roi_predictions is not None:
+            rois = roi_predictions[0]
+            for i, roi in enumerate(rois):
+                geometry = rois[i]["geometry"]
+                # TODO add validation to calculate x and y automatically
+                x = 0.5 * (geometry["left"] + geometry["right"])
```

Comment:
Why are we calculating the centers here?

Reply:
The `ImageQuery` model from the SDK requires `x` and `y` to be specified. Tyler thinks there's a way to get pydantic to do this automatically, but for now this seems easiest.

Reply:
Yeah apparently Avi asked for the sdk to return the centers of the bboxes as well as left, top, right, bottom coordinates. Tim likes this too.